aten/src/ATen/native/quantized/cpu/qpool.cpp

```diff
@@ -401,6 +402,33 @@ Tensor quantized_max_pool2d(
   return qy;
 }
 
+// TODO: when TORCH_IMPL_FUNC becomes available for quantized kernels, we can change this impl
+// to use it
+std::tuple<at::Tensor &,at::Tensor &> max_pool2d_with_indices_out_quantized_cpu(
+                       const at::Tensor & self, at::IntArrayRef kernel_size, at::IntArrayRef stride,
+                       at::IntArrayRef padding, at::IntArrayRef dilation, bool ceil_mode,
+                       at::Tensor & out, at::Tensor & indices) {
+  NoNamesGuard guard;
+
+  const int kH = safe_downcast<int, int64_t>(kernel_size[0]);
+  const int kW = kernel_size.size() == 1 ? kH : safe_downcast<int, int64_t>(kernel_size[1]);
+
+  const int dH = stride.empty() ? kH : safe_downcast<int, int64_t>(stride[0]);
+  const int dW = stride.empty() ? kW :
+                 stride.size() == 1 ? dH : safe_downcast<int, int64_t>(stride[1]);
+
+  const int padH = safe_downcast<int, int64_t>(padding[0]);
+  const int padW = padding.size() == 1 ? padH : safe_downcast<int, int64_t>(padding[1]);
+
+  const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
+  const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
+
+  max_pool2d_kernel(kCPU, out, indices, self, kW, kH, dW, dH, padW, padH, dilationW, dilationH);
+  set_quantizer_(out, make_per_tensor_affine_quantizer(self.q_scale(), self.q_zero_point(), out.scalar_type()));
```

Comment:
random question that is probably not the fault of this PR: is it really correct to `set_quantizer_` on the out argument like this? Suppose I take a view of a quantized tensor and then write into it, the quantizers of the view and the base what then become inconsistent. That seems bad!